src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -274,23 +274,116 @@ void VerifyEntryPoint(MemoryStream stream, bool expectZero)
+            CSharpCompilation comp = CreateCompilationWithMscorlib(@"
+public class C
+{
+    public int this[int i] { private get { return 0; } set { } }
+}
+");
+
+            using (var output = new MemoryStream())
+            using (var metadataOutput = new MemoryStream())
+            {
+                EmitResult emitResult = comp.Emit(output, metadataPEStream: metadataOutput,
+                    options: new EmitOptions(includePrivateMembers: false));
+                Assert.True(emitResult.Success);
+                emitResult.Diagnostics.Verify();
 
-                AssertEx.Equal(
-                    expectedMethods,
-                    ((NamedTypeSymbol)compWithMetadata.SourceModule.GetReferencedAssemblySymbols().Last()
-                        .GlobalNamespace.GetMember("C")).GetMembers().Select(m => m.ToTestDisplayString()));
+                VerifyMethods(output, new[] { "System.Int32 C.this[System.Int32 i].get", "void C.this[System.Int32 i].set",
+                    "C..ctor()", "System.Int32 C.this[System.Int32 i] { private get; set; }" });
+                VerifyMethods(metadataOutput, new[] { "void C.this[System.Int32 i].set", "C..ctor()",
+                    "System.Int32 C.this[System.Int32 i] { set; }" });
             }
         }
 
         [Fact]
+        public void EmitRefAssembly_PrivateAccessorOnEvent()
+        {
+            CSharpCompilation comp = CreateCompilationWithMscorlib(@"
+public class C
+{
+    public event System.Action PrivateAdder { private add { } remove { } }
+    public event System.Action PrivateRemover { add { } private remove { } }
+}
+");
+            comp.VerifyDiagnostics(
+                // (4,47): error CS1609: Modifiers cannot be placed on event accessor declarations
+                //     public event System.Action PrivateAdder { private add { } remove { } }
+                Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "private").WithLocation(4, 47),
+                // (5,57): error CS1609: Modifiers cannot be placed on event accessor declarations
+                //     public event System.Action PrivateRemover { add { } private remove { } }
+                Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "private").WithLocation(5, 57)
+                );
+        }
+
+        private static void VerifyMethods(MemoryStream stream, string[] expectedMethods)
+        {
+            stream.Position = 0;
+            var metadataRef = AssemblyMetadata.CreateFromImage(stream.ToArray()).GetReference();
+
+            var compWithMetadata = CreateCompilation("", references: new[] { MscorlibRef, metadataRef },
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+
+            AssertEx.Equal(
+                expectedMethods,
+                ((NamedTypeSymbol)compWithMetadata.SourceModule.GetReferencedAssemblySymbols().Last()
+                    .GlobalNamespace.GetMember("C")).GetMembers().Select(m => m.ToTestDisplayString()));
```

Comment:
Can `((NamedTypeSymbol)compWithMetadata.SourceModule...GetMember("C"))` be replaced with `compWithMetadata.GetMember<NamedTypeSymbol>("C")`?